shakenfist/config.py

```diff
@@ -14,142 +13,100 @@
+        description="how much vRAM per real unit of RAM, that is if Kernel Shared Memory (KSM) is enabled, how much to "
+                    "overcommit memory because of shared pages.")
+    RAM_SYSTEM_RESERVATION: float = Field(5.0, description="how much RAM is reserved for the OS")
+
+    # Network Options
+    FLOATING_NETWORK: IPvAnyNetwork = Field("192.168.20.0/24",
+                                            description="Network block used for NAT egress from VMs and floating IPs")
+    NETWORK_NODE_IP: IPvAnyAddress = Field(default_factory=get_node_ip,
+                                           description="IP of the node which will egress all traffic")
+
+    # Database Options
+    CLEANER_DELAY: int = Field(3600, description="How long to wait before removing old data from the database")
+
+    # Other options
+    ZONE: str = Field("shaken", description="What nova called an availability zone")
+    DOWNLOAD_URL_CIRROS: AnyHttpUrl = Field(('http://download.cirros-cloud.net/%(vernum)s/'
+                                             'cirros-%(vernum)s-x86_64-disk.img'),
+                                            description="Default download mirror for Cirros image")
+    DOWNLOAD_URL_UBUNTU: AnyHttpUrl = Field(('https://cloud-images.ubuntu.com/%(vername)s/current/'
+                                             '%(vername)s-server-cloudimg-amd64.img'),
+                                            description="Default download mirror for Ubuntu image")
+
+    #################
+    # Node Specific #
+    #################
+
+    DISK_BUS: str = Field("virtio",
+                          description="The bus to use for disk devices. One of virtio, scsi, usb, ide, etc. See libvirt"
+                                      " docs for full list of options.")
+    DISK_FORMAT: str = Field("qcow",
+                             description="The format for disks. Options are qcow (COW layers onto of image cache), "
+                                         "qcow_flat (just qcow2, no COW), and flat (just raw disk).")
+    NODE_IP: IPvAnyAddress = Field(default_factory=get_node_ip, description="IP of this node")
+    NODE_NAME: str = Field(default_factory=socket.getfqdn, description="FQDN of this node")
+    NODE_EGRESS_NIC: str = Field("eth0", description="NIC for outbound traffic")
+    STORAGE_PATH: str = Field("/srv/shakenfist", description="Where on disk instances are stored")
+
+    # Logging
+    SLOW_LOCK_THRESHOLD: float = 5.0
+    LOGLEVEL_API: str = "info"
+    LOGLEVEL_CLEANER: str = "info"
+    LOGLEVEL_MAIN: str = "info"
+    LOGLEVEL_NET: str = "info"
+    LOGLEVEL_RESOURCES: str = "info"
+    LOGLEVEL_TRIGGERS: str = "info"
+    LOG_METHOD_TRACE: int = Field(0, description="Add method name and module line number to log messages")
+
+    class Config:
+        env_prefix = 'SHAKENFIST_'
+
+    @cached_property
+    def _dict(self):
+        return self.dict()
+
+    def get(self, key):
+        return self._dict.get(key)
```

Comment:
This bit is really neat!

Reply:
Super sad, but just realized that `functools.cached_property` was introduced with python3.8, so have had to remove that